contracts/vaults/Claim.sol

```diff
@@ -33,43 +33,70 @@ contract Claim is Base {
         generalParameters.withdrawPeriod, "HVE05");
         require(_bountyPercentage <= bountyInfos[_pid].maxBounty, "HVE06");
-
-        submittedClaims[_pid] = SubmittedClaim({
+        uint256 claimId;
+        uint256 nonce;
+        do
+        {
+            claimId = uint256(keccak256(abi.encodePacked(_pid, block.number, nonce)));
+            nonce++;
+        } while (claimId == 0 || submittedClaims[claimId].beneficiary != address(0));
+        submittedClaims[claimId] = SubmittedClaim({
+            pid: _pid,
             beneficiary: _beneficiary,
             bountyPercentage: _bountyPercentage,
             committee: msg.sender,
             // solhint-disable-next-line not-rely-on-time
-            createdAt: block.timestamp
+            createdAt: block.timestamp,
+            isChallenged: false
         });
-        emit SubmitClaim(_pid, msg.sender, _beneficiary, _bountyPercentage, _descriptionHash);
+        activeClaims[_pid] = claimId;
+        emit SubmitClaim(
+            _pid,
+            claimId,
+            msg.sender,
+            _beneficiary,
+            _bountyPercentage,
+            _descriptionHash
+        );
+    }
+
+    function challenge(uint256 _claimId) onlyArbitrator external {
+        require(activeClaims[_claimId] != 0, "Can only challenge an active claim");
+        submittedClaims[_claimId].isChallenged = true;
     }
 
     /**
     * @notice Approve a claim for a bounty submitted by a committee, and transfer bounty to hacker and committee.
-    * Called only by hats governance.
-    * @param _pid The pool id
+    * callable by the  arbitrator, if isChallenged == true
+    * Callable by anyone after challengePeriod is passed and isChallenged == false
+    * @param _claimId The claim ID
     */
-    function approveClaim(uint256 _pid) external onlyOwner nonReentrant {
-        require(submittedClaims[_pid].beneficiary != address(0), "HVE10");
-        BountyInfo storage bountyInfo = bountyInfos[_pid];
-        SubmittedClaim memory submittedClaim = submittedClaims[_pid];
-        delete submittedClaims[_pid];
-
-        IERC20Upgradeable lpToken = poolInfos[_pid].lpToken;
-        ClaimBounty memory claimBounty = calcClaimBounty(_pid, submittedClaim.bountyPercentage);
-        poolInfos[_pid].balance -= claimBounty.hacker
-                            + claimBounty.hackerVested
-                            + claimBounty.committee
-                            + claimBounty.swapAndBurn
-                            + claimBounty.hackerHatVested
-                            + claimBounty.governanceHat;
+    function approveClaim(uint256 _claimId, uint256 _bountyPercentage) external nonReentrant {
+        SubmittedClaim memory submittedClaim = submittedClaims[_claimId];
+        require(
+          ((msg.sender == arbitrator && submittedClaim.isChallenged) ||
+          (submittedClaim.createdAt + challengePeriod < block.timestamp)), "HVEXX"
```

Comment:
and is the owner in the second condition

Reply:
this is as intended and also the docstring says this, anyone can approve if no-one challenges.

Reply:
what do you mean anyone? only committee can approve.. otherwise people can submit nonsense.. or i misunderstand you